e2e/registration_test.go

```diff
@@ -61,6 +62,63 @@ func TestE2E_ServiceRegistration_Defaults(t *testing.T) {
 	assert.Contains(t, check.Output, url)
 }
 
+func TestE2E_ServiceRegistration_Configured(t *testing.T) {
+	setParallelism(t)
+
+	srv := newTestConsulServer(t)
+	defer srv.Stop()
+
+	tempDir := fmt.Sprintf("%s%s", tempDirPrefix, "configured_registration")
+	cleanup := testutils.MakeTempDir(t, tempDir)
+	defer cleanup()
+
+	id := "cts-01"
+	serviceName := "cts"
+	port := testutils.FreePort(t)
+	address := fmt.Sprintf("http://127.0.0.1:%d", port)
+	rConfig := fmt.Sprintf(`
+service_registration {
+	enabled = true
+	service_name = "%s"
+	default_check {
+		enabled = true
+		address = "%s"
+	}
+}`, serviceName, address)
+
+	configPath := filepath.Join(tempDir, configFile)
+	config := baseConfig(tempDir).appendID(id).appendPort(port).
+		appendConsulBlock(srv, rConfig).appendTerraformBlock().appendDBTask()
+	config.write(t, configPath)
+
+	cmd := exec.Command("consul-terraform-sync", fmt.Sprintf("--config-file=%s", configPath))
```

Comment:
It took me a few minutes to figure out why the test was using a CTS command vs using the `ctsSetup` method. Can you add a comment just explaining that it's using a custom configuration file to test the `service_registration` parameters and can't use the default config

Something like that

Reply:
Side note, any reason not to use `api.StartCTS(t, configPath)` instead?

Reply:
That's a good point, will add comments to explain! I can't use `api.StartCTS()` because it sets the port, but I need to know what port to use beforehand so that I can configure `address`.